File: Java/SendThread.java
import java.io.IOException;


public class SendThread extends Thread {
	RxP rxp;
	String filename;
	
	public SendThread(RxP rxp, String filename){
		this.rxp = rxp;
		this.filename = filename;
	}
	// run a file sending thread
	@Override
	public void run() {
		try {
			rxp.postFile(filename);
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}

File: sendthread.py
#!/usr/bin/env python

class SendThread:

    def __init__(self, rxp, filename):
        self.rxp = rxp
        self.filename = filename

    # run a file sending thread
    def run(self, event):
        print 'start postfile'
        try:
            self.rxp.postFile(filename=self.filename, event=event)
        except IOError as e:
            print ("I/O error({0}): {1}".format(e.errno, e.strerror))